build/pcre.m4

```diff
@@ -21,8 +21,8 @@ AC_ARG_WITH(
     [test_paths="${with_pcre}"],
     [test_paths="/usr/local/libpcre /usr/local/pcre /usr/local /opt/libpcre /opt/pcre /opt /usr /opt/local"])
 
-if test "x${with_pcre2}" != "x" && test "x${with_pcre2}" != "xno"; then
-    AC_MSG_NOTICE([pcre2 specified; omitting check for pcre])
+if test "x${with_pcre}" == "x" && test "x${with_pcre}" != "xno"; then
```

Comment:
@airween the `&&` doesn't make sense here. When the first condition is true, the second condition will also always be true.
```suggestion
if test "x${with_pcre}" == "x" || test "x${with_pcre}" == "xno"; then
```